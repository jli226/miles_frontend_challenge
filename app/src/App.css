.app {
  display: flex;
  margin: 1rem;
}

.rewards-container {
  display: flex;
  border-right: 1px solid black;
  flex-direction: column;
  align-items: center;
}

.categories-container {
  display: flex;
  flex-direction: column;
  border-right: 1px solid black;
}

.reward {
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid black;
  font-size: 1rem;
  min-height: 1rem;
  margin: 1rem;
  padding: 1rem;
}

.header {
  border-bottom: 1px solid black;
  display: flex;
  justify-content: center;
  padding: 1rem 1rem;
  margin: 0 1rem;
}

.columns {
  display: flex;
  height: 100%;
}

.columns .lane:last-child {
  border-right: none;
}

.placeholder {
  height: 4.2rem;
}

.lane {
  margin-top: 1rem;
  border-right: 1px solid black;
}

.dropzone {
  padding: 0;
}

.dropzone.drop-hover {
  background-color: aqua;
}

.indicators {
  display: flex;
  flex-direction: column;
}

.indicator {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1rem;
  margin: 1rem;
  padding: 1rem;
  min-height: 1rem;
  border: 1px solid black;
  opacity: 0;
}

.indicator.belong-to {
  opacity: 1;
}
