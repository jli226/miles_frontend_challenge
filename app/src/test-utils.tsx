import React from "react";
import "mobx-react/batchingForReactDom";
import { render } from "@testing-library/react";
import { StoreProvider } from "./store/useStore";
import { DndProvider } from "react-dnd";
import { TestBackend } from "react-dnd-test-backend";

export const renderWithProviders = (children: React.ReactChild) =>
  render(
    <StoreProvider>
      <DndProvider backend={TestBackend}>{children}</DndProvider>
    </StoreProvider>
  );
