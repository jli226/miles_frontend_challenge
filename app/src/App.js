import React from 'react';
import logo from './logo.svg';
import './App.css';

const REWARDS = [
  'R1',
  'R2',
  'R3',
  'R4',
  'R5',
]

const CATEGORIES = [
  "C1",
  "C2",
  "C3",
  "C4",
  "C5",
]

function App() {
  const rewards = REWARDS.map(reward => (
    <div key={reward} className='reward'>{reward}</div>
  ))
  const categories = CATEGORIES.map(category => (
    <div className='lane'>
      <div key={category} className='header'>{category}</div>
    </div>
  ))
  return (
    <div className="app">
      <div className="rewards-container">
        <div className="header">Rewards</div>
        <div className='placeholder' />
        {rewards}
      </div>
      <div className="categories-container">
        <div className="header">Categories</div>
        <div className='columns'>
          {categories}
        </div>
      </div>
    </div>
  )
}

export default App;
