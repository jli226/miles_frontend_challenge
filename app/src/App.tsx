import React from 'react';
import './App.css';
import { useStore } from './store/useStore';
import Reward from './component/Reward';
import { observer } from 'mobx-react';
import Columns from './component/Columns';

const App = observer(() => {
  const store = useStore()
  const rewards = store.rewards.map(reward => (
    <Reward key={reward} reward={reward} />
  ))
  return (
    <div className="app">
      <div className="rewards-container">
        <div className="header">Rewards</div>
        <div className='placeholder' />
        {rewards}
      </div>
      <div className="categories-container">
        <div className="header">Categories</div>
        <Columns />
      </div>
    </div>
  )
})

export default App;
