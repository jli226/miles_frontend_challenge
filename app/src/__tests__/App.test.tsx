import React from "react";
import App from "../App";
import { renderWithProviders } from "../test-utils";

test("renders the rewards and categories", () => {
  const { container, getByText } = renderWithProviders(<App />);
  expect(getByText(/Rewards/i)).toBeInTheDocument();
  expect(getByText(/Categories/i)).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
