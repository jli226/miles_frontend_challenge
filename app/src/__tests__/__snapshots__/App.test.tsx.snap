// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the rewards and categories 1`] = `
<div>
  <div
    class="app"
  >
    <div
      class="toolbar"
    >
      <button
        disabled=""
      >
        Undo
      </button>
       
      /
       
      <button
        disabled=""
      >
        Redo
      </button>
    </div>
    <div
      class="main"
    >
      <div
        class="rewards-container"
      >
        <div
          class="header"
        >
          Rewards
        </div>
        <div
          class="placeholder"
        />
        <div
          class="reward"
        >
          R1
        </div>
        <div
          class="reward"
        >
          R2
        </div>
        <div
          class="reward"
        >
          R3
        </div>
        <div
          class="reward"
        >
          R4
        </div>
        <div
          class="reward"
        >
          R5
        </div>
      </div>
      <div
        class="categories-container"
      >
        <div
          class="header"
        >
          Categories
        </div>
        <div
          class="columns"
        >
          <div
            class="lane dropzone"
          >
            <div
              class="header"
            >
              C1
            </div>
            <div
              class="indicators"
            >
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R1
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R2
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R3
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R4
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R5
              </div>
            </div>
          </div>
          <div
            class="lane dropzone"
          >
            <div
              class="header"
            >
              C2
            </div>
            <div
              class="indicators"
            >
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R1
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R2
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R3
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R4
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R5
              </div>
            </div>
          </div>
          <div
            class="lane dropzone"
          >
            <div
              class="header"
            >
              C3
            </div>
            <div
              class="indicators"
            >
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R1
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R2
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R3
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R4
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R5
              </div>
            </div>
          </div>
          <div
            class="lane dropzone"
          >
            <div
              class="header"
            >
              C4
            </div>
            <div
              class="indicators"
            >
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R1
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R2
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R3
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R4
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R5
              </div>
            </div>
          </div>
          <div
            class="lane dropzone"
          >
            <div
              class="header"
            >
              C5
            </div>
            <div
              class="indicators"
            >
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R1
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R2
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R3
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R4
              </div>
              <div
                class="indicator"
              >
                <div
                  class="close"
                />
                R5
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
